typescript-api/scripts/generate-types.ts

```diff
@@ -51,8 +51,8 @@ async function main() {
             process.removeListener("SIGINT", onProcessExit);
         });
 
-        await new Promise(async (resolve, reject) => {
-            const onData = async (data: any) => {
+        await new Promise((resolve, reject) => {
```

Comment:
We might need to update this implementation. There is no need to create a Promise here.

Reply:
That's legacy code from moonbeam, better not to touch it. I also see a sleep(5000) that could be improved, but let's not touch code that works :joy: